### ISSUE TITLE
Running sbt 1.4.0 fails with FileAlreadyExistsException because of project/target

### ISSUE THREAD

**Issue by sideeffffect** [sbt/sbt#5970]
## steps

Run sbt launcher 1.4.0 in an environment, where there are already artifacts and other files caused by running sbt 1.3.13 before that.

## problem

Running sbt (launcher) fails with a `FileAlreadyExistsException`, most likely because `project/target` directory is already present from previous run (of sbt 1.3.13).

```
[info] [launcher] getting org.scala-sbt sbt 1.4.0  (this may take some time)...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/main-settings_2.12/1.4.0/main-settings_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/actions_2.12/1.4.0/actions_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/run_2.12/1.4.0/run_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/sbt/1.4.0/sbt-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-lang/scala-library/2.12.12/scala-library-2.12.12.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/io_2.12/1.4.0/io_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/main_2.12/1.4.0/main_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/logic_2.12/1.4.0/logic_2.12-1.4.0.jar ...
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/root/.sbt/launchers/1.4.0/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
	[SUCCESSFUL ] org.scala-sbt#logic_2.12;1.4.0!logic_2.12.jar (84ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/command_2.12/1.4.0/command_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#command_2.12;1.4.0!command_2.12.jar (29ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/collections_2.12/1.4.0/collections_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#collections_2.12;1.4.0!collections_2.12.jar (24ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/scripted-plugin_2.12/1.4.0/scripted-plugin_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#sbt;1.4.0!sbt.jar (156ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-lm-integration_2.12/1.4.0/zinc-lm-integration_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#zinc-lm-integration_2.12;1.4.0!zinc-lm-integration_2.12.jar (13ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-logging_2.12/1.4.0/util-logging_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-logging_2.12;1.4.0!util-logging_2.12.jar (27ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/launcher-interface/1.1.5/launcher-interface-1.1.5.jar ...
	[SUCCESSFUL ] org.scala-sbt#launcher-interface;1.1.5!launcher-interface.jar (10ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/github/ben-manes/caffeine/caffeine/2.8.5/caffeine-2.8.5.jar ...
	[SUCCESSFUL ] com.github.ben-manes.caffeine#caffeine;2.8.5!caffeine.jar (41ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/io/get-coursier/lm-coursier-shaded_2.12/2.0.0/lm-coursier-shaded_2.12-2.0.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#run_2.12;1.4.0!run_2.12.jar (490ms)
	[SUCCESSFUL ] org.scala-sbt#scripted-plugin_2.12;1.4.0!scripted-plugin_2.12.jar (348ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/librarymanagement-core_2.12/1.4.0/librarymanagement-core_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/librarymanagement-ivy_2.12/1.4.0/librarymanagement-ivy_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#actions_2.12;1.4.0!actions_2.12.jar (491ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/compiler-interface/1.4.0/compiler-interface-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#io_2.12;1.4.0!io_2.12.jar (493ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-compile_2.12/1.4.0/zinc-compile_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#main-settings_2.12;1.4.0!main-settings_2.12.jar (495ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-relation_2.12/1.4.0/util-relation_2.12-1.4.0.jar ...
	[SUCCESSFUL ] io.get-coursier#lm-coursier-shaded_2.12;2.0.0!lm-coursier-shaded_2.12.jar (296ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-position_2.12/1.4.0/util-position_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-position_2.12;1.4.0!util-position_2.12.jar (12ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/eed3si9n/sjson-new-scalajson_2.12/0.9.0/sjson-new-scalajson_2.12-0.9.0.jar ...
	[SUCCESSFUL ] com.eed3si9n#sjson-new-scalajson_2.12;0.9.0!sjson-new-scalajson_2.12.jar (10ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-lang/scala-reflect/2.12.12/scala-reflect-2.12.12.jar ...
	[SUCCESSFUL ] org.scala-sbt#compiler-interface;1.4.0!compiler-interface.jar (121ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/eed3si9n/sjson-new-core_2.12/0.9.0/sjson-new-core_2.12-0.9.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#zinc-compile_2.12;1.4.0!zinc-compile_2.12.jar (120ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/eed3si9n/shaded-jawn-parser_2.12/0.9.0/shaded-jawn-parser_2.12-0.9.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-relation_2.12;1.4.0!util-relation_2.12.jar (119ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/completion_2.12/1.4.0/completion_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#librarymanagement-core_2.12;1.4.0!librarymanagement-core_2.12.jar (126ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/task-system_2.12/1.4.0/task-system_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#librarymanagement-ivy_2.12;1.4.0!librarymanagement-ivy_2.12.jar (127ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/tasks_2.12/1.4.0/tasks_2.12-1.4.0.jar ...
	[SUCCESSFUL ] com.eed3si9n#shaded-jawn-parser_2.12;0.9.0!shaded-jawn-parser_2.12.jar (34ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/testing_2.12/1.4.0/testing_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#task-system_2.12;1.4.0!task-system_2.12.jar (33ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-tracking_2.12/1.4.0/util-tracking_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#completion_2.12;1.4.0!completion_2.12.jar (35ms)
	[SUCCESSFUL ] org.scala-lang#scala-library;2.12.12!scala-library.jar (653ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/jline3/jline-terminal/3.16.0-sbt-211a082ed6326908dc84ca017ce4430728f18a8a/jline-terminal-3.16.0-sbt-211a082ed6326908dc84ca017ce4430728f18a8a.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-classpath_2.12/1.4.0/zinc-classpath_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#tasks_2.12;1.4.0!tasks_2.12.jar (51ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-apiinfo_2.12/1.4.0/zinc-apiinfo_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#zinc-classpath_2.12;1.4.0!zinc-classpath_2.12.jar (31ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc_2.12/1.4.0/zinc_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-tracking_2.12;1.4.0!util-tracking_2.12.jar (36ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-control_2.12/1.4.0/util-control_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt.jline3#jline-terminal;3.16.0-sbt-211a082ed6326908dc84ca017ce4430728f18a8a!jline-terminal.jar (37ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/jline/jline/2.14.7-sbt-5e51b9d4f9631ebfa29753ce4accc57808e7fd6b/jline-2.14.7-sbt-5e51b9d4f9631ebfa29753ce4accc57808e7fd6b.jar ...
	[SUCCESSFUL ] org.scala-sbt#zinc_2.12;1.4.0!zinc_2.12.jar (14ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/jline/jline-reader/3.16.0/jline-reader-3.16.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#zinc-apiinfo_2.12;1.4.0!zinc-apiinfo_2.12.jar (29ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-interface/1.4.0/util-interface-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-control_2.12;1.4.0!util-control_2.12.jar (11ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/core-macros_2.12/1.4.0/core-macros_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#testing_2.12;1.4.0!testing_2.12.jar (60ms)
	[SUCCESSFUL ] org.scala-sbt#util-interface;1.4.0!util-interface.jar (10ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/jline/jline-terminal-jna/3.16.0/jline-terminal-jna-3.16.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/jline/jline-terminal-jansi/3.16.0/jline-terminal-jansi-3.16.0.jar ...
	[SUCCESSFUL ] org.scala-sbt.jline#jline;2.14.7-sbt-5e51b9d4f9631ebfa29753ce4accc57808e7fd6b!jline.jar (22ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-lang/scala-compiler/2.12.12/scala-compiler-2.12.12.jar ...
	[SUCCESSFUL ] org.scala-sbt#core-macros_2.12;1.4.0!core-macros_2.12.jar (14ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/jline/jline-terminal/3.16.0/jline-terminal-3.16.0.jar ...
	[SUCCESSFUL ] org.jline#jline-reader;3.16.0!jline-reader.jar (16ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/fusesource/jansi/jansi/1.18/jansi-1.18.jar ...
	[SUCCESSFUL ] org.jline#jline-terminal-jna;3.16.0!jline-terminal-jna.jar (11ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/swoval/file-tree-views/2.1.4/file-tree-views-2.1.4.jar ...
	[SUCCESSFUL ] org.scala-lang#scala-reflect;2.12.12!scala-reflect.jar (159ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/util-cache_2.12/1.4.0/util-cache_2.12-1.4.0.jar ...
	[SUCCESSFUL ] com.swoval#file-tree-views;2.1.4!file-tree-views.jar (24ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/eed3si9n/sjson-new-murmurhash_2.12/0.9.0/sjson-new-murmurhash_2.12-0.9.0.jar ...
	[SUCCESSFUL ] org.jline#jline-terminal;3.16.0!jline-terminal.jar (33ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/test-agent/1.4.0/test-agent-1.4.0.jar ...
	[SUCCESSFUL ] org.fusesource.jansi#jansi;1.18!jansi.jar (32ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/compiler-bridge_2.12/1.4.0/compiler-bridge_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#util-cache_2.12;1.4.0!util-cache_2.12.jar (22ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-classfile_2.12/1.4.0/zinc-classfile_2.12-1.4.0.jar ...
	[SUCCESSFUL ] com.eed3si9n#sjson-new-murmurhash_2.12;0.9.0!sjson-new-murmurhash_2.12.jar (11ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-core_2.12/1.4.0/zinc-core_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#test-agent;1.4.0!test-agent.jar (36ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-persist_2.12/1.4.0/zinc-persist_2.12-1.4.0.jar ...
	[SUCCESSFUL ] org.scala-sbt#compiler-bridge_2.12;1.4.0!compiler-bridge_2.12.jar (36ms)
	[SUCCESSFUL ] org.scala-sbt#zinc-core_2.12;1.4.0!zinc-core_2.12.jar (27ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-compile-core_2.12/1.4.0/zinc-compile-core_2.12-1.4.0.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/zinc-persist-core-assembly/1.4.0/zinc-persist-core-assembly-1.4.0.jar ...
	[SUCCESSFUL ] org.jline#jline-terminal-jansi;3.16.0!jline-terminal-jansi.jar (78ms)
	[SUCCESSFUL ] org.scala-sbt#zinc-classfile_2.12;1.4.0!zinc-classfile_2.12.jar (38ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/net/openhft/zero-allocation-hashing/0.10.1/zero-allocation-hashing-0.10.1.jar ...
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/sbinary_2.12/0.5.1/sbinary_2.12-0.5.1.jar ...
	[SUCCESSFUL ] com.eed3si9n#sjson-new-core_2.12;0.9.0!sjson-new-core_2.12.jar (182ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/protocol_2.12/1.4.0/protocol_2.12-1.4.0.jar ...
	[SUCCESSFUL ] net.openhft#zero-allocation-hashing;0.10.1!zero-allocation-hashing.jar(bundle) (27ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/scala-sbt/ipcsocket/ipcsocket/1.1.0/ipcsocket-1.1.0.jar ...
	[SUCCESSFUL ] org.scala-sbt.ipcsocket#ipcsocket;1.1.0!ipcsocket.jar (15ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/org/checkerframework/checker-qual/3.4.1/checker-qual-3.4.1.jar ...
	[SUCCESSFUL ] org.checkerframework#checker-qual;3.4.1!checker-qual.jar (16ms)
downloading https://artifactory.company.com/artifactory/repo1.maven.org-maven2/com/google/errorprone/error_prone_annotations/2.4.0/error_prone_annotations-2.4.0.jar ...
	[SUCCESSFUL ] com.google.errorprone#error_prone_annotations;2.4.0!error_prone_annotations.jar (10ms)
	[SUCCESSFUL ] org.scala-sbt#sbinary_2.12;0.5.1!sbinary_2.12.jar (235ms)
	[SUCCESSFUL ] org.scala-sbt#zinc-persist_2.12;1.4.0!zinc-persist_2.12.jar (239ms)
	[SUCCESSFUL ] org.scala-sbt#zinc-persist-core-assembly;1.4.0!zinc-persist-core-assembly.jar (249ms)
	[SUCCESSFUL ] org.scala-sbt#protocol_2.12;1.4.0!protocol_2.12.jar (288ms)
	[SUCCESSFUL ] org.scala-sbt#zinc-compile-core_2.12;1.4.0!zinc-compile-core_2.12.jar (300ms)
	[SUCCESSFUL ] org.scala-lang#scala-compiler;2.12.12!scala-compiler.jar (408ms)
	[SUCCESSFUL ] org.scala-sbt#main_2.12;1.4.0!main_2.12.jar (3837ms)
:: retrieving :: org.scala-sbt#boot-app
	confs: [default]
	81 artifacts copied, 0 already retrieved
[info] [launcher] getting Scala 2.12.12 (for sbt)...
:: retrieving :: org.scala-sbt#boot-scala
	confs: [default]
	6 artifacts copied, 0 already retrieved
java.nio.file.FileAlreadyExistsException: /home/tcagent/work/d25c75f9b0cc03b8/project/target
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:94)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:389)
	at java.base/java.nio.file.Files.createDirectory(Files.java:689)
	at java.base/java.nio.file.Files.createAndCheckIsDirectory(Files.java:796)
	at java.base/java.nio.file.Files.createDirectories(Files.java:742)
	at sbt.internal.BootServerSocket.<init>(BootServerSocket.java:290)
	at sbt.xMain$.getSocketOrExit(Main.scala:118)
	at sbt.xMain$.run(Main.scala:67)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:83)
	at sbt.xMain.run(Main.scala:46)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:131)
	at xsbt.boot.Launch$.run(Launch.scala:111)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
	at xsbt.boot.Launch$.launch(Launch.scala:120)
	at xsbt.boot.Launch$.apply(Launch.scala:20)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)
[error] [launcher] error during sbt launcher: java.nio.file.FileAlreadyExistsException: /home/tcagent/work/d25c75f9b0cc03b8/project/target
Process exited with code 1
```

## expectation

Sbt (launcher) should run without exceptions, even when `project/target` directory already exists.

**Comment by eatkins** [comment]
I agree that this shouldn't crash sbt but according to the documentation [Files.createDirectories](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-) only should throw if the target directory exists but is not a file. I'm curious how that could be the case here.

**Comment by sideeffffect** [comment]
@eatkins what if `project/target` is a symbolic link (linking to an actual directory)?

**Comment by eatkins** [comment]
Aah, that was an unexpected use case. Should be fixed by #5972.

**Comment by eatkins** [comment]
It should. We only care about `project/target` and even if we did care about those directories, `Files.isDirectory` follows symbolic links so we wouldn't hit that particular issue.
